### PR TITLE
feat: implement oci_tarball

### PR DESCRIPTION
**Pull request by thesayyn**
This helpful utility rule takes oci layout and makes a tarball loadable by container runtimes that support docker tarballs. works with `podman` and `docker`

### CODE REVIEW

**Comment by cgrindel** [comment]
@thesayyn Are you looking for a review on this PR? If so, could you select some folks and get the process going?

**Comment by cgrindel** [comment]
I will convert this to a draft. Please mark it as ready for review and assign some reviewers when it is readay.

File: oci/private/tarball.bzl [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+example;
+
+```shell
+bazel build //target
+docker load --input $(bazel cquery --output=files //target)
+```
+"""
+
+attrs = {
+    "image": attr.label(mandatory = True, allow_single_file = True, doc = "Label to an oci_image target"),
```
Comment (alexeagle) [inline oci/private/tarball.bzl:23]: do you have the pre-commit installed? I would have expected this to be formatted differently

File: oci/private/tarball.bzl [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    "_tarball_sh": attr.label(allow_single_file = True, default = "//oci/private:tarball.sh.tpl"),
+    "_build_tar": attr.label(
+        default = Label("@rules_pkg//pkg/private/tar:build_tar"),
+        cfg = "exec",
+        executable = True,
+        allow_files = True,
+    ),
+}
+
+# TODO(thesayyn): find a way to call pkg_tar without depending on the private stuff.
```
Comment (alexeagle) [inline oci/private/tarball.bzl:34]: should this point to an issue on rules_pkg?

**Review by alexeagle — APPROVED** [review]
can you add any tests / executable example to show that it's working?

File: oci/private/tarball.bzl (hunk 1 above)
Comment (thesayyn) [reply oci/private/tarball.bzl:24]: I have it but still didn't format the line to be multi.

**Comment by thesayyn** [comment]
> can you add any tests / executable example to show that it's working?

already there in the `example/js` folder. and asserted as part of the CI workflow with docker as well. 

https://github.com/bazel-contrib/rules_oci/pull/30/files#diff-02978fcc5af423efbd83422b15debe41c50f942b49dd3e4de963ab714ee5f37eR38-R42

https://github.com/bazel-contrib/rules_oci/pull/30/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR45-R47

**Pull request merged**